osu.Game/Collections/DrawableCollectionListItem.cs

```diff
@@ -117,11 +116,67 @@ private partial class ItemTextBox : OsuTextBox
         {
             protected override float LeftRightPadding => item_height / 2;
 
+            private const float count_text_size = 12;
+
+            [Resolved]
+            private RealmAccess realm { get; set; } = null!;
+
+            private readonly Live<BeatmapCollection> collection;
+
+            private OsuSpriteText countText = null!;
+
+            private IDisposable? itemCountSubscription;
+
+            public ItemTextBox(Live<BeatmapCollection> collection)
+            {
+                this.collection = collection;
+
+                CornerRadius = item_height / 2;
+            }
+
             [BackgroundDependencyLoader]
             private void load(OsuColour colours)
             {
                 BackgroundUnfocused = colours.GreySeaFoamDarker.Darken(0.5f);
                 BackgroundFocused = colours.GreySeaFoam;
+
+                if (collection.IsManaged)
+                {
+                    TextContainer.Height *= (Height - count_text_size) / Height;
+                    TextContainer.Margin = new MarginPadding { Bottom = count_text_size };
+
+                    TextContainer.Add(countText = new OsuSpriteText
+                    {
+                        Anchor = Anchor.BottomLeft,
+                        Origin = Anchor.TopLeft,
+                        Depth = float.MinValue,
+                        Font = OsuFont.Default.With(size: count_text_size, weight: FontWeight.SemiBold),
+                        Margin = new MarginPadding { Top = 2, Left = 2 },
+                        Colour = colours.Yellow
+                    });
+
+                    itemCountSubscription = realm.SubscribeToPropertyChanged(r => r.Find<BeatmapCollection>(collection.ID), c => c.BeatmapMD5Hashes, _ =>
+                        Scheduler.AddOnce(() =>
+                        {
+                            int count = collection.PerformRead(c => c.BeatmapMD5Hashes.Count);
+
+                            countText.Text = count == 1
+                                // Intentionally not localised until we have proper support for this (see https://github.com/ppy/osu-framework/pull/4918
+                                // but also in this case we want support for formatting a number within a string).
+                                ? $"{count:#,0} beatmap"
+                                : $"{count:#,0} beatmaps";
```

Comment:
As discussed in https://github.com/ppy/osu/pull/23000, "beatmaps" is incorrect here. I would suggest "items" if we going with the same direction here.